# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmCode.cpp
# Commit: b37cddf0a3b3
# Full Hash: b37cddf0a3b33cc340398dd1698367b3ddc9f771
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-03-13 10:29:34
# Description:
#   Bug 1952215 - wasm: Fix calleePtr for lazy stubs. r=rhunt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240945
# ==============================================================================

diff -r b1f1c0da3306 -r b37cddf0a3b3 js/src/wasm/WasmCode.cpp
--- a/js/src/wasm/WasmCode.cpp	Wed Mar 12 16:16:17 2025 +0000
+++ b/js/src/wasm/WasmCode.cpp	Wed Mar 12 16:16:56 2025 +0000
@@ -461,7 +461,7 @@
   WasmMacroAssembler masm(alloc);
 
   const FuncExportVector& funcExports = tierCodeBlock.funcExports;
-  uint8_t* segmentBase = tierCodeBlock.segment->base();
+  uint8_t* codeBase = tierCodeBlock.base();
 
   CodeRangeVector codeRanges;
   DebugOnly<uint32_t> numExpectedRanges = 0;
@@ -471,7 +471,7 @@
     // Exports that don't support a jit entry get only the interp entry.
     numExpectedRanges += (funcType.canHaveJitEntry() ? 2 : 1);
     void* calleePtr =
-        segmentBase + tierCodeBlock.codeRange(fe).funcUncheckedCallEntry();
+        codeBase + tierCodeBlock.codeRange(fe).funcUncheckedCallEntry();
     Maybe<ImmPtr> callee;
     callee.emplace(calleePtr, ImmPtr::NoCheckToken());
     if (!GenerateEntryStubs(masm, funcExportIndex, fe, funcType, callee,
